plugins/api/discovery/src/org/apache/cloudstack/discovery/ApiDiscoveryServiceImpl.java

```diff
@@ -236,7 +238,9 @@ private ApiDiscoveryResponse getCmdRequestMap(Class<?> cmdClass, APICommand apiC
                 boolean isAllowed = true;
                 for (APIChecker apiChecker : _apiAccessCheckers) {
                     try {
-                        apiChecker.checkAccess(user, apiName);
+                        if (!"ApiRateLimitServiceImpl".equals(apiChecker.getName())) {
```

Comment:
you can use something like this.class.getName() or the <class>.getName()... instead of the hard coded string comparison.

Reply:
Using the 'ApiRateLimitServiceImpl' class instead of a hard coded string introduces the rate limit plugin modules as a dependency to the api discovery module.  I didn't think that was a good idea.